packages/lesswrong/components/ea-forum/EAEmojisHeader.tsx

```diff
@@ -0,0 +1,584 @@
+import React, { FC, MouseEvent, ReactNode, createContext, useCallback, useContext, useEffect, useRef, useState } from "react";
+import { Components, registerComponent } from "../../lib/vulcan-lib";
+import { AnalyticsContext } from "../../lib/analyticsEvents";
+import { isClient } from "../../lib/executionEnvironment";
+import { EMOJIS_HEADER_HEIGHT } from "../common/Header";
+import { useCurrentUser } from "../common/withUser";
+import { Link } from "../../lib/reactRouterWrapper";
+import { gql, useMutation, useQuery } from "@apollo/client";
+import ForumNoSSR from "../common/ForumNoSSR";
+import classNames from "classnames";
+import { userIsAdminOrMod } from "../../lib/vulcan-users";
+import { useLoginPopoverContext } from "../hooks/useLoginPopoverContext";
+
+if (isClient) {
+  require("emoji-picker-element");
```

Comment:
This library provides the emoji picker as a web component that we wrap for use in react below - we shouldn't import it on the server as it relies on browser specific stuff